include/wil/result_originate.h

```diff
@@ -59,18 +60,19 @@ namespace wil
 
                 if (shouldOriginate)
                 {
+                    auto message = wil::make_unique_string_nothrow<wil::unique_hstring>(failure.pszMessage);

```

Comment:
This should probably be a hstring_reference as well - avoid the duplication/allocation until some later time that RoOriginate needs to preserve it.

Otherwise, fully agree with the change.

Reply:
I don't think I can directly use hstring_reference as that would assume cppwinrt?

Reply:
There is the raw win32 method that underlies it [WindowsCreateStringReference](https://learn.microsoft.com/en-us/windows/win32/api/winstring/nf-winstring-windowscreatestringreference).  If WRL can be assumed (it probably cannot) then you could also use [WRL::Wrappers::HStringReference](https://learn.microsoft.com/en-us/cpp/cppcx/wrl/hstringreference-class?view=msvc-170).

Reply:
Yeah, looks like there's not a `wil::create_hstring_reference` like cppwinrt's, yet...